### PR TITLE
Support new s6 overlay

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

Need the new base image! **Don't merge before!!!**

## Type of change

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by mdegat01** [comment]
Interesting. Is this a necessary change or an improvement? I'm asking mostly because nearly all addons have an identical finish script for their services. So is this a change nearly all addon developers will have to make as well to upgrade to the latest base image?

**Comment by pvizeli** [comment]
Yeah, if we update an add-on, we have to update the finished script if they have one. I will write a blog post on the dev

File: rootfs/etc/services.d/supervisor/finish [hunk 1]
```diff
@@ -5,4 +5,4 @@
 if { s6-test ${1} -ne 100 }
 if { s6-test ${1} -ne 256 }
 
-redirfd -w 2 /dev/null s6-svscanctl -t /var/run/s6/services
+/run/s6/basedir/bin/halt
```
Comment (agners) [inline rootfs/etc/services.d/supervisor/finish:8]: The old code also redirected exit code from what I understand. Is the new line doing the same?

Comment (pvizeli) [reply rootfs/etc/services.d/supervisor/finish:8]: It does the same; both don't redirect the exit code. If you want that, you can do it but that also change

**Comment by frenck** [comment]
> Interesting. Is this a necessary change or an improvement? 

If we want to keep upgrading s6-overlay, it is necessary (We can't halt upgrading forever though).

There are possible more changes needed, however, the currently s6-overlay is currently backward compatible to an extent that keeps current needed changes minimal.

I think that currently having an upgrade path with minimal changes is good to get the ball rolling; after that, we can look into how (and if) we upgrade/migrate other things (e.g., maybe we should migrate from `services.d` to the s6-rc, consider if we need to migrate `cont-init.d` scripts to oneshot services with dependencies, etc).

**Review by mdegat01 — APPROVED** [review]

**Pull request merged**